h2o-core/src/main/java/water/parser/CsvParser.java

```diff
@@ -459,17 +459,19 @@
   }
 
   @Override protected int fileHasHeader(byte[] bits, ParseSetup ps) {
-    boolean hasHdr = true;
```

Comment:
Why did the default change from doesn't have header to has header?

Reply:
The logic is inverted. If there is at least one of the columns with the same name as in the original file, it is considered to have a header (previous state was in case of a perfect match across all columns).
In the "else" branch, hasHdr is set to true explicitely to keep the original behavior.